File: js/commands/update.js
import { spawnSync } from 'node:child_process';

const INSTALL_COMMAND = 'curl -fsSL https://aynig.org/install.sh | bash';

export function registerUpdateCommand(program) {
  program
    .command('update')
    .description('Download and install the latest AYNIG release')
    .action(() => {
      const result = spawnSync('sh', ['-c', INSTALL_COMMAND], {
        stdio: 'inherit'
      });

      if (result.status !== 0) {
        process.exit(result.status ?? 1);
      }
    });
}

File: go/internal/commands/update.go
package commands

import (
	"fmt"
	"os"
	"os/exec"
)

const installCommand = "curl -fsSL https://aynig.org/install.sh | bash"

func Update() error {
	cmd := exec.Command("sh", "-c", installCommand)
	cmd.Stdout = os.Stdout
	cmd.Stderr = os.Stderr
	cmd.Stdin = os.Stdin

	if err := cmd.Run(); err != nil {
		return fmt.Errorf("update failed: %w", err)
	}

	return nil
}
